### PR TITLE
F/wpc 880 add comment context

### PR DESCRIPTION
**Pull request by michael-scheurer**
This PR is about the implementation of the comment context / history. I placed the same history component twice:

- In the comment view
image-20221107-123537

- in the article metadata
image-20221107-123609

### CODE REVIEW

**Comment by michael-scheurer** [comment]
This pr implements a comment history in the comment edit view as well as in the article metadata view.

image

File: packages/editor/src/client/atoms/comment/commentHistory.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+interface CommentHistoryProps {
+  commentItemType: CommentItemType
+  commentItemID: string
+  commentId?: string
+}
+
+export function CommentHistory({commentId, commentItemType, commentItemID}: CommentHistoryProps) {
+  const {t} = useTranslation()
+  const [comments, setComments] = useState<FullCommentFragment[] | undefined>()
+  const {data, refetch} = useCommentListQuery({
```
Comment (Itrulia) [inline packages/editor/src/client/atoms/comment/commentHistory.tsx:53]: I guess this should use a LazyQuery as the useEffect below is called immediately

**Review by Itrulia — CHANGES_REQUESTED** [review]
Works amazing and looks fine (I have a question about a lazy query but it's non blocking for me). The only thing I found is that this now breaks the layout at default zoom level on my MacBook when the context is expanded. Screenshots are below

Screenshot 2022-11-09 at 10 31 19
Screenshot 2022-11-09 at 10 31 22

**Review by penk85 — COMMENTED** [review]
Looks really good and useful feature. I'd suggest maybe a clear visual indication for the current comment besides open drop down (maybe highlighted border or bold and to put other comments in gray text, possibly smaller.) . I also wonder if the edit text box could go inside the comment preview to simplify the ui.

File: packages/editor/src/client/atoms/comment/commentHistory.tsx [hunk 2]
```diff
@@ -10,44 +10,65 @@ import {
...
   }
   const childComments = comments.filter(tmpComment => tmpComment.parentComment?.id === comment.id)
   return (
     <div style={{marginTop: '20px', borderLeft: '1px lightgrey solid', paddingLeft: '20px'}}>
       {childComments.map(childComment => (
-        <div key={childComment.id}>
-          <CommentPreview comment={childComment} expanded={childComment.id === originCommentId} />
+        <div
+          key={childComment.id}
+          id={childComment.id === originComment?.id ? 'current-comment' : ''}>
```
Comment (Itrulia) [inline packages/editor/src/client/atoms/comment/commentHistory.tsx:38]: Could this result in multiple 'current-comment' ids?

**Review by penk85 — COMMENTED** [review]
:+1:

**Review by penk85 — APPROVED** [review]

**Review by Itrulia — APPROVED** [review]

**Pull request merged**